### ISSUE TITLE
Exe not working on Windows XP (Trac #77)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#81]
When run from the command line I get the error:

"The system cannot execute the specified program."

When run by clicking on the exe, I get the error box (screenshot attached) message:

"C:\Documents and Settings\User\My Documents\dev\microdrop\dist\microdrop\microdrop.exe

This application has failed to start because the application configuration is incorrect. Reinstalling the application may fix this problem."

Tested on v0.1-408-g4fb6983.

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/77
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "When run from the command line I get the error:\n\n\"The system cannot execute the specified program.\"\n\nWhen run by clicking on the exe, I get the error box (screenshot attached) message:\n\n\"C:\\Documents and Settings\\User\\My Documents\\dev\\microdrop\\dist\\microdrop\\microdrop.exe\n\nThis application has failed to start because the application configuration is incorrect. Reinstalling the application may fix this problem.\"\n\nTested on v0.1-408-g4fb6983.", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Exe not working on Windows XP", 
    "priority": "blocker", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-03-16T23:41:34", 
    "milestone": "Microdrop 1.0", 
    "owner": "cfobel", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120316T23:41:59: ryan changed attachment from "" to "error msg.jpg"

**Comment by cfobel** [comment]
Trac update at 20120317T00:39:45: ryan commented:

> The same msi file (built on XP buildbot) ran successfully on one Win 7 computer, but on another, it gave the following error:
> 
> $ ./microdrop.exe
> Found embedded PKG: C:\Program Files (x86)\microdrop\microdrop.exe
> Extracting binaries
> manifestpath: C:\Program Files (x86)\microdrop\microdrop.exe.manifest
> Error activating the context
> C:\Program Files (x86)\microdrop\python27.dll
> Manipulating evironment
> PYTHONPATH=C:/Program Files (x86)/microdrop
> importing modules from CArchive
> extracted iu
> extracted struct
> extracted archive
> Installing import hooks
> outPYZ1.pyz
> Running scripts
> Traceback (most recent call last):
>  File "<string>", line 31, in <module>
>  File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
> 1.pyz/os", line 150, in makedirs
>  File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
> 1.pyz/os", line 157, in makedirs
> WindowsError: [Error 5] Access is denied: 'C:/Program Files (x86)/microdrop\\sup
> port'
> RC: -1 from win32comgenpy
> OK.
> 
> Not sure if the two problems are related.

**Comment by cfobel** [comment]
Trac update at 20120317T01:12:05: ryan commented:

> This looks like an unrelated problem. Opening a new ticket.
> 
> Replying to [comment:1 ryan]:
> > The same msi file (built on XP buildbot) ran successfully on one Win 7 computer, but on another, it gave the following error:
> > 
> > $ ./microdrop.exe
> > Found embedded PKG: C:\Program Files (x86)\microdrop\microdrop.exe
> > Extracting binaries
> > manifestpath: C:\Program Files (x86)\microdrop\microdrop.exe.manifest
> > Error activating the context
> > C:\Program Files (x86)\microdrop\python27.dll
> > Manipulating evironment
> > PYTHONPATH=C:/Program Files (x86)/microdrop
> > importing modules from CArchive
> > extracted iu
> > extracted struct
> > extracted archive
> > Installing import hooks
> > outPYZ1.pyz
> > Running scripts
> > Traceback (most recent call last):
> >  File "<string>", line 31, in <module>
> >  File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
> > 1.pyz/os", line 150, in makedirs
> >  File "C:\buildbot\microdrop\WinXP_Py2_7\build\build\pyi.win32\microdrop\outPYZ
> > 1.pyz/os", line 157, in makedirs
> > WindowsError: [Error 5] Access is denied: 'C:/Program Files (x86)/microdrop\\sup
> > port'
> > RC: -1 from win32comgenpy
> > OK.
> > 
> > Not sure if the two problems are related.
>

**Comment by cfobel** [comment]
Trac update at 20120328T21:41:48: 
* christian@fobel.net commented:

> commit 6cfbe94ffb43dd6fca5587224094e9962de6945e
> Author: Christian Fobel <christian@fobel.net>
> Date:   Wed Mar 28 17:38:35 2012 -0400
> 
>     Remove import of PIL, it was causing broken EXE
>     
>     Fixes #81.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20120328T21:52:13: cfobel commented:

> The installer is also confirmed to work in the Windows XP virtual machine image as of 6cfbe94ffb43dd6fca5587224094e9962de6945e.

**Comment by cfobel** [comment]
Trac update at 20120329T16:57:38: 
* ryan commented:

> Commit 6cfbe94ffb43dd6fca5587224094e9962de6945e allows the exe to run on XP, but it breaks video on both Windows XP and Windows 7.


* ryan changed resolution from "fixed" to ""
* ryan changed status from "closed" to "reopened"

**Comment by cfobel** [comment]
Trac update at 20120329T17:16:51: 
* ryan changed attachment from "" to "runtime error.jpg"
* ryan commented:

> Runtime Error message

**Comment by cfobel** [comment]
Trac update at 20120329T17:17:50: 
* ryan changed attachment from "" to "microdrop.spec.1"
* ryan commented:

> microdrop.spec file version 1

**Comment by cfobel** [comment]
Trac update at 20120329T17:21:37: 
* ryan changed attachment from "" to "microdrop.spec.2"
* ryan commented:

> microdrop.spec file version 2

**Comment by cfobel** [comment]
Trac update at 20120329T17:26:24: ryan commented:

> I tried a couple of variations to the microdrop.spec file to see if I could resolve this problem (tested on Windows 7). With no modifications (v0.1-421-gf3c959a), the file: "PIL._imaging.pyd" is included in the distribution directory. The application starts with no errors, but the video does not work.
> 
> When I use microdrop.spec.1 (attached), the "PIL._imaging.pyd" file is no longer in the distribution directory, but there is now a "PIL" subdirectory. The attached runtime error is displayed twice on startup and the video does not work.
> 
> Using microdrop.spec.2 (attached), both the "PIL" subdirectory and the "PIL._imaging.pyd" are present in the distribution folder. I also see the runtime error twice on startup and video does not work.

**Comment by cfobel** [comment]
Trac update at 20120329T20:49:05: ryan commented:

> It seems to be a problem with the PIL manifest file:
> 
> https://groups.google.com/forum/?fromgroups#!topic/pyinstaller/xNe1fvOlv4s
> 
> If we reinclude the PIL imports that were taken out in cfbe94ffb43dd6fca5587224094e9962de6945e and then remove the following lines from microdrop.exe.manifest, everything works on both XP and Win 7:
> 
> <dependency>
>     <dependentAssembly>
>       <assemblyIdentity type="win32" name="Microsoft.VC90.DebugCRT" version="9.0.21022.8" processorArchitecture="x86" publicKeyToken="1fc8b3b9a1e18e3b"></assemblyIdentity>
>     </dependentAssembly>
>   </dependency>
> 
> Just investigating the best way to resolve this problem (either rebuilding PIL or downgrading to 1.1.6).

**Comment by cfobel** [comment]
Trac update at 20120329T21:17:21: ryan commented:

> So PIL 1.1.6 doesn't have a Python 2.7 release. Looks like we either need to rebuild the PIL package or patch it after it is installed with the modified _imagingft.pyd (attached).

**Comment by cfobel** [comment]
Trac update at 20120329T21:30:48: ryan changed attachment from "" to "_imagingft.pyd"

**Comment by cfobel** [comment]
Trac update at 20120330T00:46:59: ryan commented:

> I repackaged PIL-1.1.7.win32-py2.7.exe after replacing the _imagingft.pyd file. It is available as a git repository at:
> 
> http://microfluidics.utoronto.ca/git/PIL.git
> 
> and the executable can be downloaded here:
> 
> http://microfluidics.utoronto.ca/software/PIL-1.1.7.win32-py2.7.exe

**Comment by cfobel** [comment]
Trac update at 20120330T01:23:56: 
* ryan@fobel.net commented:

> commit ec06ebb83b79aa14786175ca9fe13536c3f498e2
> Author: Ryan Fobel <ryan@fobel.net>
> Date:   Thu Mar 29 20:42:38 2012 -0400
> 
>     Fix #81
>     
>     Update install_dependencies to point to repackaged version of PIL.
>     Re-add the PIL imports to microdrop.py file to make video work again
>     for pyinstaller version.


* ryan@fobel.net changed resolution from "" to "fixed"
* ryan@fobel.net changed status from "reopened" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "core" to "microdrop"
